geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/registry/ProtobufOperationContextRegistry.java

```diff
@@ -109,5 +111,12 @@ private void addContexts() {
             opsResp -> ClientProtocol.Response.newBuilder().setGetAvailableServersResponse(opsResp),
             new ResourcePermission(ResourcePermission.Resource.CLUSTER,
                 ResourcePermission.Operation.READ)));
+
+    operationContexts.put(RequestAPICase.HANDSHAKEREQUEST,
+        new ProtobufOperationContext<>(ClientProtocol.Request::getHandshakeRequest,
+            new HandshakeRequestOperationHandler(),
+            opsResp -> ClientProtocol.Response.newBuilder().setHandshakeResponse(opsResp),
+            new ResourcePermission(ResourcePermission.Resource.DATA,
+                ResourcePermission.Operation.READ)));
```

Comment:
I don't like that this is never used and is bypassed by the state processor. Perhaps we could leave a comment to that effect? Maybe mark it "ALL" here?

Reply:
That sounds like a great plan, but unfortunately there isn't an ALL option for resource or operation fields in ResourcePermissions.